### PR TITLE
[Windows] Replace Docker EE with Docker CE

### PR DESCRIPTION
**Pull request by miketimofeev**
# Description
Support for DockerMsftProvider ends at the end of September and will be fully transferred to Mirantis. The alternative is Docker CE / Moby.
This PR replaces installation of Docker EE in favor of Docker CE and also
- Changes docker-compose v2 installation to `C:\ProgramData\docker` as all the docker configs and libs are now placed there
- Changes `wincred` installation to the same directory as Docker CE executables (win\system32) as there is no more `C:\Program Files\Docker` in the path
- Adds Docker to the exclusion list in Software report blank version tester as the version doesn't match the general format — `master-dockerproject-2022-03-26`

#### Related issue:
https://github.com/actions/runner-images/issues/6181

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by mikhailkoliada — APPROVED** [review]

**Review by erik-bershel — DISMISSED** [review]

File: images/win/scripts/Installers/Install-Docker.ps1
Comment (al-cheb) [inline images/win/scripts/Installers/Install-Docker.ps1:11]: nitpick: replace to Start-DownloadWithRetry or use full param names(Output,Uri,etc)

Comment (miketimofeev) [reply images/win/scripts/Installers/Install-Docker.ps1]: Makes sense, replaced, thank you!

**Review by al-cheb — APPROVED** [review]

**Review by ddobranic — APPROVED** [review]

**Review by igorboskovic3 — APPROVED** [review]

**Comment by pawelpabich** [comment]
@miketimofeev just making sure you are aware of  https://github.com/microsoft/Windows-Containers/issues/256

**Comment by miketimofeev** [comment]
> @miketimofeev just making sure you are aware of  https://github.com/microsoft/Windows-Containers/issues/256

Thanks! I'll take a closer look

**Comment by miketimofeev** [comment]
@pawelpabich I took a closer look and it seems that those binaries at https://master.dockerproject.org/ were updated last time on March, 27 so they are not nightly builds after all (at least for now). I would rather wait for the decision from Windows-Containers maintainers than implementing a script to download the latest stable version from https://download.docker.com/win/static/stable/x86_64/

**Pull request merged**
